server/rosetta/client_online.go

```diff
@@ -0,0 +1,396 @@
+	blockResp, err := c.BlockByHeight(ctx, height)
+	if err != nil {
+		return crgtypes.BlockTransactionsResponse{}, err
+	}
+
+	txs, err := c.listTransactionsInBlock(ctx, blockResp.Block.Index)
+	if err != nil {
+		return crgtypes.BlockTransactionsResponse{}, err
+	}
+
+	return crgtypes.BlockTransactionsResponse{
+		BlockResponse: blockResp,
+		Transactions:  sdkTxsWithHashToRosettaTxs(txs),
+	}, nil
+}
+
+// Coins fetches the existing coins in the application
+func (c *Client) coins(ctx context.Context) (sdk.Coins, error) {
+	supply, err := c.bank.TotalSupply(ctx, &bank.QueryTotalSupplyRequest{})
+	if err != nil {
+		return nil, crgerrs.FromGRPCToRosettaError(err)
+	}
+	return supply.Supply, nil
+}
+
+// listTransactionsInBlock returns the list of the transactions in a block given its height
+func (c *Client) listTransactionsInBlock(ctx context.Context, height int64) ([]*sdkTxWithHash, error) {
+	txQuery := fmt.Sprintf(`tx.height=%d`, height)
+	txList, err := c.clientCtx.Client.TxSearch(ctx, txQuery, true, nil, nil, "")
+	if err != nil {
+		return nil, crgerrs.WrapError(crgerrs.ErrUnknown, err.Error())
+	}
+
+	sdkTxs, err := tmResultTxsToSdkTxsWithHash(c.clientCtx.TxConfig.TxDecoder(), txList.Txs)
+	if err != nil {
+		return nil, err
+	}
+	return sdkTxs, nil
+}
+
+func (c *Client) TxOperationsAndSignersAccountIdentifiers(signed bool, txBytes []byte) (ops []*types.Operation, signers []*types.AccountIdentifier, err error) {
+	txConfig := c.getTxConfig()
+	rawTx, err := txConfig.TxDecoder()(txBytes)
+	if err != nil {
+		return nil, nil, err
+	}
+	txBldr, _ := txConfig.WrapTxBuilder(rawTx)
+
+	var accountIdentifierSigners []*types.AccountIdentifier
+	if signed {
+		addrs := txBldr.GetTx().GetSigners()
+		for _, addr := range addrs {
+			signer := &types.AccountIdentifier{
+				Address: addr.String(),
+			}
+			accountIdentifierSigners = append(accountIdentifierSigners, signer)
+		}
+	}
+
+	return sdkTxToOperations(txBldr.GetTx(), false, false), accountIdentifierSigners, nil
+}
+
+// GetTx returns a transaction given its hash
+func (c *Client) GetTx(_ context.Context, hash string) (*types.Transaction, error) {
+	txResp, err := authclient.QueryTx(c.clientCtx, hash)
+	if err != nil {
+		return nil, crgerrs.WrapError(crgerrs.ErrUnknown, err.Error())
+	}
+	var sdkTx sdk.Tx
+	err = c.ir.UnpackAny(txResp.Tx, &sdkTx)
+	if err != nil {
+		return nil, crgerrs.WrapError(crgerrs.ErrCodec, err.Error())
+	}
+	return sdkTxWithHashToOperations(&sdkTxWithHash{
+		HexHash: txResp.TxHash,
+		Code:    txResp.Code,
+		Log:     txResp.RawLog,
+		Tx:      sdkTx,
+	}), nil
+}
+
+// GetUnconfirmedTx gets an unconfirmed transaction given its hash
+// NOTE(fdymylja): not implemented yet
+func (c *Client) GetUnconfirmedTx(_ context.Context, _ string) (*types.Transaction, error) {
+	return nil, crgerrs.WrapError(crgerrs.ErrNotImplemented, "get unconfirmed transaction method is not supported")
```

Comment:
why not query mempool's unconfirmed and check for equal hash? You can check up until an arbitrary number of txs if you want (say 30)

Reply:
We did not supported it because there was other priorities then like making it work for all the 3 versions.

Due to the very small block time and because in the rosetta spec they don't require this to be a complete implementation.

Testing this endpoint was a little bit difficult too, do you know if it is possible to set the block creation time bigger so we have the time to check this endpoint?